code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,298 @@
+/obj/item/inflatable
+	name = "inflatable"
+	w_class = 2
+	icon = 'icons/obj/inflatable.dmi'
+	var/deploy_path = null
+
+/obj/item/inflatable/attack_self(mob/user)
+	if(!deploy_path)
+		return
+	playsound(loc, 'sound/items/zip.ogg', 75, 1)
+	user << "<span class='notice'>You inflate \the [src].</span>"
+	var/obj/structure/inflatable/R = new deploy_path(user.loc)
+	src.transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	qdel(src)
+
+/obj/item/inflatable/wall
+	name = "inflatable wall"
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation."
+	icon_state = "folded_wall"
+	deploy_path = /obj/structure/inflatable/wall
+
+/obj/item/inflatable/door/
+	name = "inflatable door"
+	desc = "A folded membrane which rapidly expands into a simple door on activation."
+	icon_state = "folded_door"
+	deploy_path = /obj/structure/inflatable/door
+
+/obj/structure/inflatable
+	name = "inflatable"
+	desc = "An inflated membrane. Do not puncture."
+	density = 1
+	anchored = 1
+	opacity = 0
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "wall"
+	penetration_dampening = 3
+
+	var/undeploy_path = null
+	var/health = 50.0
+
+/obj/structure/inflatable/wall
+	name = "inflatable wall"
+	undeploy_path = /obj/item/inflatable/wall
+
+/obj/structure/inflatable/New(location)
+	..()
+	update_nearby_tiles(need_rebuild=1)
+
+/obj/structure/inflatable/Destroy()
+	update_nearby_tiles()
+	..()
+
+/obj/structure/inflatable/Cross(atom/movable/mover, turf/target, height=0, air_group=0)
```

Comment:
I'm not going to pretend I know how it works, on bay it's 
```
/obj/structure/inflatable/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
	return 0
```
and it was one of the lines giving errors, I looked at either our walls or windows and saw they looked similar guessing it was what blocked air